### PR TITLE
Summary QC

### LINKED ISSUE #843: Feature Request: QC function to summarize the QC
It would be beneficial to add in a function that summarizes the QC in an object (0=good, 1=indeterminate, 2=Bad)...

### PR DESCRIPTION
**Pull request by kenkehoe**
This PR contains a new method to create the Standardized QC Summary file. It will convert current embedded QC which is bit packed to integer/flag QC that contains a single test per assessment. Plan is for ADC Data Discovery to use this code to generate QC Summary files on the fly for user order. This is part of ARM TSK0312159.

- [x] Closes #843
- [x] Tests added
- [x] Documentation reflects changes
- [x] PEP8 Standards or use of linter
- [x] Xarray Dataset or DataArray variable naming follows 'ds' or 'da' naming

### CODE REVIEW

**Comment by kenkehoe** [comment]
Don't integrate the PR yet. I need to discuss current implementation with Adam and Michael first.

File: act/qc/qc_summary.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
+import numpy as np
```
Comment (zssherman) [inline act/qc/qc_summary.py:1]: Can we add a general docstring summary above the imports for this new module

File: act/qc/qc_summary.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+        cleanup_qc : boolean
+            Call clean.cleanup() method to convert to standardized ancillary quality control
+            variables. The quality control summary requires the current embedded quality
+            control variables to use ACT standards.
+
+        Returns
+        -------
+        return_ds : xarray.Dataset
+            ACT Xarray dataset with quality control variales converted to summary flag values.
```
Comment (zssherman) [inline act/qc/qc_summary.py:33]: variales to variables

**Review by zssherman — CHANGES_REQUESTED** [review]

**Comment by zssherman** [comment]
@kenkehoe Thanks for the PR! I added some review comments

File: act/qc/qc_summary.py (hunk 1 above)
Comment (mgrover1) [reply act/qc/qc_summary.py:1]: Agree here

File: act/qc/qc_summary.py (hunk 2 above)
Comment (mgrover1) [reply act/qc/qc_summary.py:33]: Agree here - small typo

**Review by mgrover1 — CHANGES_REQUESTED** [review]
Just a couple small suggestions - same as Zach

File: act/qc/qc_summary.py (hunk 1 above)
Comment (kenkehoe) [reply act/qc/qc_summary.py:1]: Done.

File: act/qc/qc_summary.py (hunk 2 above)
Comment (kenkehoe) [reply act/qc/qc_summary.py:33]: Done.

**Review by mgrover1 — APPROVED** [review]
thanks for the quick response @kenkehoe

**Review by AdamTheisen — APPROVED** [review]

**Comment by zssherman** [comment]
@kenkehoe Cool with me merging this? I saw Adam approved it

File: act/io/arm.py [hunk 3]
```diff
@@ -248,7 +248,7 @@ def read_arm_netcdf(
             file_dates.append(pts[2])
             file_times.append(pts[3])
         else:
-            if ds['time'].size > 1:
+            if ds['time'].ndim > 0:
```
Comment (zssherman) [inline act/io/arm.py:251]: Can we keep this as size? ndim is being depreciated in xarray

File: act/qc/clean.py [hunk 4]
```diff
@@ -749,6 +756,28 @@ def clean_arm_qc(
...
+        try:
+            text = 'DQMS is composed of a suite of tests which includes SERI QC'
+            SERI_QC = text in self._ds.attrs['comment']
+        except KeyError:
+            SERI_QC = False
+
+        if SERI_QC and global_qc is None and qc_attributes is None:
+            self._ds.clean.clean_seri_qc()
+
+        # If the QC was not cleaned up because it is not correctcly formatted with
```
Comment (zssherman) [inline act/qc/clean.py:770]: correctcly to correctly

File: act/qc/qc_summary.py
Comment (zssherman) [inline act/qc/qc_summary.py:1]: I noticed this wasn't added to the init, if we can add that for imports and documentation generation.

File: tests/qc/test_qc_summary.py [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+    qc_ma = result.qcfilter.get_masked_data(var_name, rm_assessments='Incorrect')
+    assert np.sum(np.where(qc_ma.mask)[0]) == 13560
+
+    assert np.sum(np.where(result[qc_var_name].values == 0)) == 884575
+    qc_ma = result.qcfilter.get_masked_data(var_name, rm_assessments='Not failing')
+    assert np.sum(np.where(qc_ma.mask)[0]) == 884575
+
+
+@pytest.mark.big
+@pytest.mark.skipif('ARCHIVE_DATA' not in environ, reason="Running outside ADC system.")
```
Comment (zssherman) [inline tests/qc/test_qc_summary.py:108]: CI seems to be failing as its not finding or giving a KeyError for 'ARCHIVE_DATA'.

**Review by zssherman — CHANGES_REQUESTED** [review]

File: tests/qc/test_qc_summary.py (hunk 5 above)
Comment (kenkehoe) [reply tests/qc/test_qc_summary.py:108]: Yeah, it's a temporary fix to get the test to run locally while I wait for the new test data PR to go through.

File: act/qc/clean.py (hunk 4 above)
Comment (kenkehoe) [reply act/qc/clean.py:770]: @zssherman  let's hold off on merging. I'm doing some additional testing. Should I cancel this PR and submit a new one soon when it is ready to merge? Is there a way to temporarily stop it from running tests?

File: tests/qc/test_qc_summary.py (hunk 5 above)
Comment (zssherman) [reply tests/qc/test_qc_summary.py:108]: @kenkehoe Ah makes sense

File: act/qc/clean.py (hunk 4 above)
Comment (zssherman) [reply act/qc/clean.py:770]: Sounds good! And the tests should be stopped now.

**Comment by zssherman** [comment]
@kenkehoe You cool with me merging this?

**Comment by kenkehoe** [comment]
@zssherman hold off on merger. I found one more issue and I need to run some more testing. Hopefully this will be figured out soon.

**Comment by zssherman** [comment]
@kenkehoe No worries at all!

**Comment by AdamTheisen** [comment]
@kenkehoe how's this looking for merging?

@zssherman have your requested changes been implemented?  Can we resolve that request?

**Comment by kenkehoe** [comment]
@AdamTheisen I believe this ready to merge. I have tested the crap out of this and it is working pretty well.

**Review by zssherman — APPROVED** [review]

**Comment by zssherman** [comment]
@kenkehoe Thanks for doing those changes! Merging

**Pull request merged**